ecr_helper/helper.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+#
+# Copyright 2018 Google, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
```

Comment:
You don't need [as big a comment as this one](https://github.com/knative/build-templates/blob/master/gcr_helper/helper.sh#L17), but a quick overview of how this works (it looks like it fetches a 12-hour token with the user's current credentials) would be helpful.

Does the ECR authorization token have any permissions/access to other AWS resources? I'm presuming it has the ability to act as the current logged in user against ECR (delete, re-tag). Might be worth calling that out.

Reply:
added comments